### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by chambliss**
Hey Nina!

This is just an intermediate PR - I haven't addressed all of the feedback you gave me yet (like continuing to build out the README). I'm pushing this one because I'd love it if you could run the train_model.py script on the GPU and push the results back here (I have it set up such that it should save model and optimizer state_dicts, f1/loss, etc after each epoch). I have it set to do just 2 epochs for now, but feel free to raise it if you think it'd be a good idea to do more. I tried to do a single-epoch run on my machine to test it out like we discussed, but my puny CPU segfaulted. Oops 😅 

I did also install pre-commit hooks w/ black and flake8 configured, but I'm not sure if it worked. I'm going to check/test it out some more this weekend. 

To summarize, this PR includes:
- Restructuring repo, creating base main.py file to build from 
- Fixed issues with preprocessing of tags (en data) not working correctly (in same commit as above)
- Setup of pre-commit hooks and pre-commit config
- train_model.py file for training the English PyTorch model

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: stable
+    hooks:
+    - id: black
+      files: 'python/*'
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (ninalopatina) [inline .pre-commit-config.yaml]: This is a good start, but you want to make sure everything is being checked, including yaml files. yaml is a separate hook id

File: python/utils/train_model.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    for step, batch in enumerate(train_dataloader):
+
+        print('Starting a new training batch.')
+
+        # add batch to gpu
+        batch = tuple(t.to(device) for t in batch)
+        b_input_ids, b_input_mask, b_labels = batch
+
+        # forward pass
+        loss = model(b_input_ids, token_type_ids=None,
```
Comment (ninalopatina) [inline python/utils/train_model.py]: this needs to be "loss, something_else" because of an update to the output of model()

File: python/utils/train_model.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        batch = tuple(t.to(device) for t in batch)
+        b_input_ids, b_input_mask, b_labels = batch
+
+        with torch.no_grad():
+            tmp_eval_loss = model(b_input_ids, token_type_ids=None,
+                                  attention_mask=b_input_mask, labels=b_labels)
+            logits = model(b_input_ids, token_type_ids=None,
+                           attention_mask=b_input_mask)
+
+        logits = logits.detach().cpu().numpy()
```
Comment (ninalopatina) [inline python/utils/train_model.py]: error here: Train loss: 0.1815053676309194
Starting validation loop.
Traceback (most recent call last):
 File "train_model.py", line 281, in <module>
   logits = logits.detach().cpu().numpy()
AttributeError: 'tuple' object has no attribute 'detach'
